_studio/mfx_lib/vpp/src/mfx_vpp_sw_core.cpp

```diff
@@ -1016,22 +989,24 @@ mfxStatus VideoVPPBase::Query(VideoCORE * core, mfxVideoParam *in, mfxVideoParam
         }
 
         /* [OUT VPP] data */
-        if( out->vpp.Out.FourCC != MFX_FOURCC_YV12 &&
+        if (out->vpp.Out.FourCC != MFX_FOURCC_YV12 &&
             out->vpp.Out.FourCC != MFX_FOURCC_NV12 &&
             out->vpp.Out.FourCC != MFX_FOURCC_YUY2 &&
             out->vpp.Out.FourCC != MFX_FOURCC_RGB4 &&
+#if defined (MFX_ENABLE_FOURCC_RGB565)
+            out->vpp.Out.FourCC != MFX_FOURCC_RGB565 &&
+#endif
+#if defined(MFX_VA_LINUX)
             out->vpp.Out.FourCC != MFX_FOURCC_UYVY &&
-#ifdef MFX_ENABLE_RGBP
-            out->vpp.Out.FourCC != MFX_FOURCC_RGBP &&
 #endif
             out->vpp.Out.FourCC != MFX_FOURCC_P010 &&
             out->vpp.Out.FourCC != MFX_FOURCC_P210 &&
 #if (MFX_VERSION >= 1027)
             out->vpp.Out.FourCC != MFX_FOURCC_Y210 &&
             out->vpp.Out.FourCC != MFX_FOURCC_Y410 &&
 #endif
-            out->vpp.Out.FourCC != MFX_FOURCC_AYUV &&
-            out->vpp.Out.FourCC != MFX_FOURCC_A2RGB10 )
+            out->vpp.In.FourCC != MFX_FOURCC_AYUV &&
```

Comment:
this is intentional correction? comment on line 991 says this block is dedicated to OUT params

Reply:
It is my mistake due to inattention; corrected